src/gdb_packet.c

```diff
@@ -60,27 +60,29 @@ char *gdb_packet_buffer(void)
 #endif /* EXTERNAL_PACKET_BUFFER */
 
 /* https://sourceware.org/gdb/onlinedocs/gdb/Packet-Acknowledgment.html */
-void gdb_set_noackmode(bool enable)
+void gdb_set_noackmode(const bool enable)
 {
-	/*
-	 * If we were asked to disable NoAckMode, and it was previously enabled,
-	 * it might mean we got a packet we determined to be the first of a new
-	 * GDB session, and as such it was not acknowledged (before GDB enabled NoAckMode),
-	 * better late than never.
-	 *
-	 * If we were asked after the connection was terminated, sending the ack will have no effect.
-	 */
-	if (!enable && noackmode)
-		gdb_if_putchar(GDB_PACKET_ACK, true);
-
 	/* Log only changes */
 	if (noackmode != enable)
 		DEBUG_GDB("%s NoAckMode\n", enable ? "Enabling" : "Disabling");
 
 	noackmode = enable;
 }
 
+bool gdb_noackmode(void)
+{
+	return noackmode;
+}
+
 #ifndef DEBUG_GDB_IS_NOOP
+/**
+ * To debug packets from the perspective of GDB we can use the following command:
+ *     set debug remote 1
```

Comment:
Technically `set debug remote on`, but we suspect this doesn't work right on older GDBs. Please check if you can. Don't worry changing it if it is broken on older GDBs, otherwise it'd be nice to document the intended way.

Reply:
Right, it'd be nice to see where this is officially documented, but I don't think it is anywhere? Might peek the source code to check

Reply:
It's briefly documented in the source docs, in the description they mention mostly `on` / `off` but the internal tests always use `1` / `0`, and it is implied both should work. Anyway... I replaced with `on` because it looks nicer, this was more for me than anything honestly, because I always forget the command and then have a hard time finding it again...

Reply:
`set debug remote on` does not work on 2020 release gdb-9.1 and likely others (older), printing an unrelated message `No symbol table is loaded.  Use the "file" command.`, or `No symbol "on" in current context`. Recommend documenting both. Change is dated January 2021.

Reply:
> Might peek the source code to check

https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=02349803fc5a7941b4e23d1580bd8b600d6be1bd

Reply:
Argh, I only had the source distribution with me, no git info, so I couldn't look back at the changes...

Anyway, I don't feel like changing this again and it serves the purpose nonetheless, whoever is debugging this code in the future is bound to know what they are doing...